mathjax3-ts/input/tex/color/ColorConfiguration.ts

```diff
@@ -0,0 +1,376 @@
+ */
+function get_RGB(rgb: string): string {
+    const rgbParts: string[] = rgb.trim().split(/\s*,\s*/);
+    let RGB = '#';
+
+    if (rgbParts.length !== 3) {
+        throw new TexError('ModelArg1', 'Color values for the %1 model require 3 numbers', 'RGB');
+    }
+
+    for (const rgbPart of rgbParts) {
+        if (!rgbPart.match(/^\d+$/)) {
+            throw new TexError('InvalidNumber', 'Invalid number');
+        }
+
+        const n = parseInt(rgbPart);
+        if (n > 255) {
+            throw new TexError('ModelArg2',
+                'Color values for the %1 model must be between %2 and %3',
+                'RGB', '0', '255');
+        }
+
+        let pn = n.toString(16);
+        if (pn.length < 2) {
+            pn = '0' + pn;
+        }
+        RGB += pn;
+    }
+    return RGB;
+}
+
+/**
+ * Get a gray-scale value.
+ * 
+ * @param {string} gray The color definition in RGB: `0.5`.
+ * @return {string} The color definition in CSS format e.g. `#808080`.
+ */
+function get_gray(gray: string): string {
+    if (!gray.match(/^\s*(\d+(\.\d*)?|\.\d+)\s*$/)) {
+        throw new TexError('InvalidDecimalNumber', 'Invalid decimal number');
+    }
+
+    const n: number = parseFloat(gray);
+    if (n < 0 || n > 1) {
+        throw new TexError('ModelArg2',
+            'Color values for the %1 model must be between %2 and %3',
+            'gray', '0', '1');
+    }
+    let pn = Math.floor(n * 255).toString(16);
+    if (pn.length < 2) {
+        pn = '0' + pn;
+    }
+
+    return `#${pn}${pn}${pn}`;
+}
+
+/**
+ * Get a named color.
+ *
+ * @param {string} name The color definition in RGB: `128,0,255`.
+ * @return {string} The color definition in CSS format e.g. `#44ff00`.
+ */
+function get_named(name: string): string {
+    if (COLORS.has(name)) {
+        return COLORS.get(name);
+    } else {
+        return name;
+    }
+}
+
+export let ColorMethods: Record<string, ParseMethod> = {};
+
+
+/**
+ * Override \color macro definition.
+ *
+ * @param {TexParser} parser The calling parser.
+ * @param {string} name The name of the control sequence.
+ */
+ColorMethods.Color = function (parser: TexParser, name: string) {
+    const model = parser.GetBrackets(name, '');
+    const color = getColor(model, parser.GetArgument(name));
+
+    const style = parser.itemFactory.create('style')
+                        .setProperties({styles: { mathcolor: color }});
+    parser.stack.env['color'] = color;
+
+    parser.Push(style);
+};
+
+
+/**
+ * Define the \textcolor macro.
+ *
+ * @param {TexParser} parser The calling parser.
+ * @param {string} name The name of the control sequence.
+ */
+ColorMethods.TextColor = function (parser: TexParser, name: string) {
+    const model = parser.GetBrackets(name, '');
+    const color = getColor(model, parser.GetArgument(name));
+    const old = parser.stack.env['color'];
+
+    parser.stack.env['color'] = color;
+    const math = ParseUtil.internalMath(parser, parser.GetArgument(name));
```

Comment:
Despite its name, `\textcolor` does *not* switch into text mode in LaTeX, so your original was correct here (using `ParseArg()` rather than `internalMath()`).  The contents of `\textcolor` should be processed in math-mode.

Reply:
Thanks @dpvc. There's something I don't understand here, I need to explore more and see how can I address your feedback.